Doc/library/imaplib.rst

```diff
@@ -30,12 +30,14 @@ Three classes are provided by the :mod:`imaplib` module, :class:`IMAP4` is the
 base class:
 
 
-.. class:: IMAP4(host='', port=IMAP4_PORT)
+.. class:: IMAP4(host='', port=IMAP4_PORT, timeout=None)
```

Comment:
Is timeout=0 supported? Non-blocking socket.

If not, we should either document that it's not supported or even raise an exception if it's equal to zero ("if timeout is not None and not timeout: ...").

Reply:
Since [socket.makefile](https://docs.python.org/3.9/library/socket.html#socket.socket.makefile) does not support Non-blocking socket.
timeout 0 should not be accepted.

Thanks for letting me know :)